src/netaddress.h

```diff
@@ -111,6 +111,8 @@ static constexpr size_t ADDR_INTERNAL_SIZE = 10;
 /// SAM 3.1 and earlier do not support specifying ports and force the port to 0.
 static constexpr uint16_t I2P_SAM31_PORT{0};
 
+std::string OnionToString(Span<const uint8_t> addr);
```

Comment:
Since this is now public, it would be nice to document it. Something like:

```cpp
/**
 * Generate the ".onion" address which corresponds to a public key.
 * @param[in] addr public key bytes, must be 32.
 * @returns for example 5g72ppm3krkorsfopcm2bi7wlv4ohhs4u4mlseymasn7g7zhdcyjpfid.onion
 */
```